server/channels/api4/user_local.go

```diff
@@ -233,27 +233,25 @@ func localGetUsers(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func localGetUsersByIds(c *Context, w http.ResponseWriter, r *http.Request) {
-	userIDs := model.ArrayFromJSON(r.Body)
-
-	if len(userIDs) == 0 {
+	userIDs, err := model.ArrayFromJSONLimited(r.Body, *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes)
+	if err != nil {
+		c.Err = model.NewAppError("localGetUsersByIds", model.PayloadParseError, nil, "", http.StatusBadRequest).Wrap(err)
+		return
+	} else if len(userIDs) == 0 {
 		c.SetInvalidParam("user_ids")
 		return
 	}
 
-	// we remove the duplicate IDs as it can bring a significant load to the
-	// database.
-	userIDs = model.RemoveDuplicateStrings(userIDs)
-
 	sinceString := r.URL.Query().Get("since")
 
 	options := &store.UserGetByIdsOpts{
 		IsAdmin: c.IsSystemAdmin(),
 	}
 
 	if sinceString != "" {
-		since, err := strconv.ParseInt(sinceString, 10, 64)
-		if err != nil {
-			c.SetInvalidParamWithErr("since", err)
+		since, err2 := strconv.ParseInt(sinceString, 10, 64)
+		if err2 != nil {
+			c.SetInvalidParamWithErr("since", err2)
```

Comment:
Small nit: `parseErr` would be my preference here naming wise rather than numbering.